### PR TITLE
Sema: Fix source compatibility break with default initialization of optional properties

### PR DESCRIPTION
**Pull request by slavapestov**
Fixes rdar://problem/35319847.

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

File: lib/Sema/TypeCheckDecl.cpp [hunk 1]
```diff
@@ -4127,8 +4127,19 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
             TC.checkTypeModifyingDeclAttributes(var);
 
           // Decide whether we should suppress default initialization.
-          if (!PBD->isDefaultInitializable(i))
-            continue;
+          //
+          // Note: in Swift 4, we screwed up: a property of Optional<Int> does
```
Comment (lattner) [inline lib/Sema/TypeCheckDecl.cpp]: Please don't add comments like this that tell a personal story of the team.  Instead keep it factual:

// Note: The Swift 4 compiler contained a bug that ...

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Pull request merged**